src/rdkafka_broker.c

```diff
@@ -1567,6 +1569,111 @@ int rd_kafka_socket_cb_generic (int domain, int type, int protocol,
 }
 
 
+
+/**
+ * @brief Update the reconnect backoff.
+ *        Should be called when a connection is made.
+ *
+ * @locality broker thread
+ * @locks none
+ */
+static void
+rd_kafka_broker_update_reconnect_backoff (rd_kafka_broker_t *rkb,
+                                          const rd_kafka_conf_t *conf,
+                                          rd_ts_t now) {
+        int backoff;
+
+        /* If last connection attempt was more than reconnect.backoff.max.ms
+         * ago, reset the reconnect backoff to the initial
+         * reconnect.backoff.ms value.
+         * Or if the current backoff increment has reached the max value. */
+        if (rkb->rkb_ts_reconnect + (conf->reconnect_backoff_max_ms * 1000) <
+            now ||
+            rkb->rkb_reconnect_backoff_ms >= conf->reconnect_backoff_max_ms)
+                rkb->rkb_reconnect_backoff_ms = conf->reconnect_backoff_ms;
```

Comment:
Do we really want to reset to the minimum backoff once enough time has elapsed?  It seems better to keep the backoff high until we have some contact with the server.

Reply:
You're probably right!